devito/types/grid.py

```diff
@@ -165,6 +165,12 @@ def origin(self):
         """Physical coordinates of the domain origin."""
         return self._origin
 
+    @property
+    def origin_map(self):
+        """Map between origin symbols and their values"""
+        origin_pos = [i.data for i in self.origin]
+        return dict(zip(self.origin, origin_pos))
```

Comment:
Can one-line
`{o: o.data for o in self.origin}`

Reply:
Good point, will do